test/sequence/us_core_guidance/us_core_r4_patient_read_only_sequence_test.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require_relative '../../test_helper'
+
+describe Inferno::Sequence::USCoreR4PatientReadOnlySequence do
+  before do
+    @base_url = 'http://www.example.com/fhir'
+    @token = 'ABC'
+    @patient_id = '123'
+    @sequence_class = Inferno::Sequence::USCoreR4PatientReadOnlySequence
+    @client = FHIR::Client.new(@base_url)
+    @client.set_bearer_token(@token)
+    @instance = Inferno::Models::TestingInstance.create
+    @instance.patient_id = @patient_id
+  end
+
+  describe 'unauthenticated read test' do
+    before do
+      @test = @sequence_class[:unauthenticated_read]
+      @sequence = @sequence_class.new(@instance, @client)
+    end
+
+    it 'skips if no token is set' do
+      assert_raises(Inferno::SkipException) { @sequence.run_test(@test) }
+    end
+
+    it 'fails when the server does not return a 401' do
+      @instance.token = 'ABC'
+      stub_request(:get, "#{@base_url}/Patient/#{@patient_id}")
+        .to_return(status: 200)
+
+      exception = assert_raises(Inferno::AssertionException) { @sequence.run_test(@test) }
+
+      assert_equal exception.message, 'Bad response code: expected 401, but found 200'
+    end
+
+    it 'succeeds when the server returns a 401' do
+      @instance.token = @token
+      stub_request(:get, "#{@base_url}/Patient/#{@patient_id}")
+        .to_return(status: 401)
+
+      @sequence.run_test(@test)
+    end
+  end
+
+  describe 'authenticated read test' do
+    before do
+      @test = @sequence_class[:authenticated_read]
+      @sequence = @sequence_class.new(@instance, @client)
+      @instance.token = @token
+    end
+
+    it 'fails when the server does not return a 200' do
+      stub_request(:get, "#{@base_url}/Patient/#{@patient_id}")
+        .with(headers: { 'Authorization' => "Bearer #{@instance.token}" })
+        .to_return(status: 202)
+
+      exception = assert_raises(Inferno::AssertionException) { @sequence.run_test(@test) }
+
+      assert_equal exception.message, 'Bad response code: expected 200, 201, but found 202. '
```

Comment:
Seems weird that on a read failure our warning would indicate that a `201`, which is for `Created`, is allowed.

Should we have separate `assert_ok` functionality for read and create?  We might just be able to modify the `assert_ok` method if the response it is passed has enough information (I think it does include the request as well).

Reply:
I updated `assert_response_unauthorized` (which previously took a 401 or 406) in this PR because it was an easy change to make that seemed very unlikely to break anything. `assert_response_ok` currently takes a 200 or a 201, which I don't think we want, but it's used far more often than `assert_response_unauthorized`, and changing it is more likely to break something, so I think it deserves its own ticket.

Reply:
Agreed, let's make a new ticket for this.